vendor/github.com/Dataman-Cloud/swan-resolver/nameserver/record_generator.go

```diff
@@ -19,24 +19,31 @@ func (r rrs) add(name, host string) bool {
 	if host == "" {
 		return false
 	}
+	var v []string
 	v, ok := r[name]
 	if !ok {
-		v = make(map[string]struct{})
+		v = append(v, host)
 		r[name] = v
 	} else {
-		// don't overwrite existing values
-		_, ok = v[host]
-		if ok {
-			return false
+		hostDuplicated := false
```

Comment:
how about add a util method SliceContaints([]string, string) bool, if r[name] contains host, then skip the process.